core/src/test/java/org/elasticsearch/indices/analyze/AnalyzeActionIT.java

```diff
@@ -503,4 +503,20 @@ public void testDetailAnalyzeWithMultiValuesWithCustomAnalyzer() throws Exceptio
 
     }
 
+    public void testNonExistTokenizer() {
+        assertAcked(prepareCreate("test").addAlias(new Alias("alias")));
+        ensureGreen();
+
+        try {
+            AnalyzeResponse analyzeResponse = client().admin().indices()
+                .prepareAnalyze("this is a test")
+                .setAnalyzer("not_exist_analyzer")
+                .get();
+            fail("shouldn't get here");
+        } catch (Throwable t) {
+            assertThat(t, instanceOf(IllegalArgumentException.class));
```

Comment:
It is a big deal! Otherwise we have no idea if this test is hitting a completely different error than what was expected. Testing exceptions should always check something in the exception message to be sure the *right* exception was caught (we use IAE all over the place, so the exception class alone is not sufficient).